### PR TITLE
Fix #synth edge-case, minor refactor with Stack type

### PR DESCRIPTION
**Pull request by dselsam**

### CODE REVIEW

**Comment by leodemoura** [comment]
Could you please rename the commits to 
```
fix: handle no instances for a goal
feat:  stack wrapper around array
```

File: library/Init/Data/Stack/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Simple stack API implemented using an array.
+-/
+prelude
+import Init.Data.Array
+import Init.Data.Int
+universes u v w
+
+structure Stack (α : Type u) :=
+(vals : Array α := Array.empty)
+
```
Comment (leodemoura) [inline library/Init/Data/Stack/Basic.lean]: We can now write `#[]` for `Array.empty`

Comment (dselsam) [reply library/Init/Data/Stack/Basic.lean]: Noted.

**Comment by dselsam** [comment]
> Could you please rename the commits to
> 
> ```
> fix: handle no instances for a goal
> feat:  stack wrapper around array
> ```

Done. Force-pushed.

**Comment by leodemoura** [comment]
merged

**Pull request closed without merging**